### ISSUE TITLE
MedSAM2_pretrain.pth (pretrained weights) and which .yaml file?

### ISSUE THREAD

**Issue by Konnos998** [MedicineToken/Medical-SAM2#68]
Hello there,

I try to get familiar with the model and I wanted to know, if for example you do:

**python train_2d.py -net sam2 -exp_name REFUGE_MedSAM2 -vis 1 -sam_ckpt ./checkpoints/sam2_hiera_small.pt -sam_config sam2_hiera_s -image_size 1024 -out_size 1024 -b 4 -val_freq 1 -dataset REFUGE -data_path ./data/REFUGE**

the logical thing is to put -sam_ckpt -> sam2_hiera_small.pt and -sam_config -> sam2_hiera_s (which I think is the .yaml file from sam2_train folder). Also in this folder I donwloaded 2 more files from sam2 model (sam2_hiera_l.yaml and sam2_hiera_base_plus.yaml) because they were missing...I don't know if this is correct. Now, my question is, if you do:

**python train_2d.py -net sam2 -exp_name REFUGE_MedSAM2 -vis 1 -sam_ckpt ./checkpoints/MedSAM2_pretrain.pth -sam_config ??????????? -image_size 1024 -out_size 1024 -b 4 -val_freq 1 -dataset REFUGE -data_path ./data/REFUGE**

what do you put in place of the question marks? Nothing, or one of the existing .yaml files in the sam2_train folder?

**Comment by Konnos998** [comment]
Also -train_vis 1 (to visualize the training) doesn't work for some reason...What I can see is (which I see it anyway with the default command) is the validation and epoch proccesses. I don't know what more is shown with the -train_vis set to 1, but I can't see it.

**Comment by piyushw0203** [comment]
were you able to figure anything out?

**Comment by Konnos998** [comment]
Hello brother!
I've red that medsam2 is based on Sam 2, which is on github too. I went to Sam2's link on github, went through the folders and also downloaded sam2_hiera_l.yaml, sam2_hiera_b+.yaml (that were missing from the sam2_train folder, of medsam2 model). Also, in the checkpoints folder I have all the .pt files (small, tiny, large, base_plus) and I also have the pretrained weights MedSAM2_pretrained.pth (I found it on the readme file of medsam2 on github). Also, I try to make the model to stop with ctrl+C and then continue from where it left (like a checkpoint) and I've noticed that when you resume training you shouldn't put the pretrained weights again, because it gets confused and the dice index goes from 0.87 for example, to 0.17. Sometimes this happens even if you do it all right. I'm trying to understand the logic behind, and the only solution may be to just study about neural networks...😬
I'm runnning it on Windows 10, through WSL2 and VS Code, on a laptop with an RTX 2060 6GB mobile, and I had to do some modifications on the train_2d.py and write os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:512" to reduce the load on the gpu by allocating the memory. I'm new to this, I don't even know if what I say is 100% right...
